### PR TITLE
refactor(csv): Rely on `http.Error` in 'pkg/csv'

### PR DESCRIPTION
**Pull request by rukletsov**
## Description

This PR teaches `csv.WriteError()` to deduce the status code from the error class und ensures appropriate HTTP headers are set.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [ ] ~Evaluated and added CHANGELOG entry if required~
- [ ] ~Determined and documented upgrade steps~
- [ ] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

### Here I tell how I validated my change

Unit tests and CI run.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/cve/image/csv/handler.go [hunk 1]
```diff
@@ -111,14 +111,14 @@ func ImageCVECSVHandler() http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		query, rQuery, err := parser.ParseURLQuery(r.URL.Query())
 		if err != nil {
-			csv.WriteError(w, http.StatusBadRequest, err)
+			csv.WriteErrorWithCode(w, http.StatusBadRequest, err)
```
Comment (0x656b694d) [inline central/cve/image/csv/handler.go:114]: Our sentinel `errox.InvalidArgs` maps to `http.StatusBadRequest`. Do we want to say:
```suggestion
			csv.WriteError(w, errox.InvalidArgs.CausedBy(err))
```
to add "invalid arguments: " prefix to the error parser message?

**Review by 0x656b694d — APPROVED** [review]
Couple of comments for consideration, otherwise LGTM.

File: central/cve/image/csv/handler.go (hunk 1 above)
Comment (rukletsov) [reply central/cve/image/csv/handler.go:114]: I guess for me the biggest value of sentinels is not that extra "invalid arguments: " prefix but error class determination at a point where it's known best. However, your suggestion might help get rid of `WriteErrorWithCode()` and hence increase the visibility of sentinels.

File: pkg/csv/utils.go [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
 // Utility functions to be used for CSV exporting.
 
-// WriteError Writes the given error to the given http.ResponseWriter.
-func WriteError(w http.ResponseWriter, code int, err error) {
-	w.Header().Set("Content-Type", "text/plain")
-	w.WriteHeader(code)
-	_, _ = fmt.Fprint(w, err)
+// WriteError responds with the error message and HTTP status code deduced from
+// the error class. Appropriate response headers are set.
+func WriteError(w http.ResponseWriter, err error) {
```
[the remark this answers is not in the document]
Comment (rukletsov) [reply pkg/csv/utils.go:16]: Technically it depends on how the underlying response writer flushes but adding a comment won't hurt, thanks!

**Review by 0x656b694d — APPROVED** [review]
Thanks!

**Pull request merged**